react/react1/week3/vite-project/src/todo-details.jsx

```diff
@@ -0,0 +1,70 @@
+import { useParams } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import { useRef, useState } from "react";
+import { useEffect } from "react";
+
+const TodoDetails = () => {
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const [todo, setTodo] = useState(null);
+  const form = useRef(null);
+
+  useEffect(() => {
+    fetch(`http://localhost:8000/todoList/${id}`)
+      .then((result) => result.json())
+      .then((res) => {
+        setTodo(res);
+      });
+  }, [id]);
+
+  const handleEdit = (e) => {
+    debugger;
+    e.preventDefault();
+    const bodyRequest = Object.fromEntries(
+      new FormData(form.current).entries()
+    );
+    fetch(`http://localhost:8000/todoList/${id}`, {
+      method: "PUT",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        ...todo,
+        ...bodyRequest,
+      }),
+    }).then(()=>{
+      navigate('/')
+    });
+  };
+
+  const template = todo ? (
+    <form
+      ref={form}
+      key={todo.id}
+      className="todo-details"
+      onSubmit={handleEdit}
+    >
+      <label>
+        Title : 
+        <input type="text" name="title" defaultValue={todo.title} />
```

Comment:
Make your form controlled. If we give the inputs a value props and an unChanged prop, we don't have to use a reference to the form, and we can change the todo state as we change the input values